src/user.rs

```diff
@@ -13,144 +15,231 @@ use async_trait::async_trait;
     ///
-    /// # Returns
-    /// Id of deleted device or IronOxideErr
-    async fn user_delete_device(&self, device_id: Option<&DeviceId>) -> Result<DeviceId>;
-
-    /// Verify a user given a JWT for their user record.
+    /// Returns a `None` if the user could not be found.
     ///
     /// # Arguments
-    /// - `jwt` - Valid IronCore JWT
-    /// - `timeout` - timeout for this operation or None for no timeout. Timed out operations return IronOxideErr::OperationTimedOut
+    /// - `jwt`     - Valid IronCore or Auth0 JWT
+    /// - `timeout` - Timeout for this operation or `None` for no timeout
     ///
-    /// # Returns
-    /// Option of whether the user's account record exists in the IronCore system or not. Err if the request couldn't be made.
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let jwt = "";
+    /// let verify_result = IronOxide::user_verify(jwt, None).await?;
+    /// let user_id = verify_result.expect("User not found!").account_id();
+    /// # Ok(())
+    /// # }
+    /// ```
     async fn user_verify(
         jwt: &str,
         timeout: Option<std::time::Duration>,
     ) -> Result<Option<UserResult>>;
 
-    /// Get a list of user public keys given their IDs. Allows discovery of which user IDs have keys in the
-    /// IronCore system to determine of they can be added to groups or have documents shared with them.
+    /// Lists all of the devices for the current user.
+    ///
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// let devices_result = sdk.user_list_devices().await?;
+    /// let devices: Vec<UserDevice> = devices_result.result().to_vec();
+    /// # Ok(())
+    /// # }
+    /// ```
+    async fn user_list_devices(&self) -> Result<UserDeviceListResult>;
+
+    /// Gets users' public keys given their IDs.
+    ///
+    /// Allows discovery of which user IDs have keys in the IronCore system to help determine if they can be added to groups
+    /// or have documents shared with them.
+    ///
+    /// Only returns users whose keys were found in the IronCore system.
     ///
     /// # Arguments
     /// - users - List of user IDs to check
     ///
-    /// # Returns
-    /// Map from user ID to users public key. Only users who have public keys will be returned in the map.
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # use std::convert::TryFrom;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// let user1 = UserId::try_from("colt")?;
+    /// let user2 = UserId::try_from("fake_user")?;
+    /// // This will only return one entry, for user "colt"
+    /// let get_result = sdk.user_get_public_key(&[user1, user2]).await?;
```

Comment:
Should we provide code here to match the output with the input?

Reply:
I'm not exactly sure what you're suggesting. Could you expand a bit?

Reply:
The docs say that this call "allows discovery of which user IDs have keys in the IronCore system". I think we should consider showing what that looks like. If I send in a list of 10 users, and 5 of them don't have public keys, we can easily demonstrate how to get lists of valid and invalid users.

I didn't compile this, but something like:
```rust
 let (invalid_users, valid_users) = users.iter().partition(|u| !get_result.contains_key(u));
```
If you used itertools' partition you could return the valid user's public keys.

Reply:
I like it. Hopefully my implementation is reasonable

Reply:
looks good.